src/structures/Guild.js

```diff
@@ -332,6 +332,39 @@ class Guild {
   }
 
   /**
+   * The data for a role
+   * @typedef {Object} AddGuildMemberOptions
+   * @property {string} accessToken An oauth2 access token granted with the guilds.join to the bot's application
+   * for the user you want to add to the guild
+   * @property {string} [nick] Value to set users nickname to
+   * @property {Collection<Snowflake, Role>|Role[]|string[]} [roles] The roles or role IDs to add
+   * @property {boolean} [mute] If the user is muted
+   * @property {boolean} [deaf] If the user is deafened
+   */
+
+  /**
+   * Add a user to this guild using OAuth2
+   * @param {UserResolvable|string} user The user or ID of the user to add to guild
+   * @param {AddGuildMemberOptions} options Options object containing the access_token
+   * @returns {Promise<GuildMember>}
+   */
+  addMember(user, options) {
+    if (options.roles) {
+      var roles = options.roles;
+      if (roles instanceof Collection || (roles instanceof Array && roles[0] instanceof Role)) {
+        options.roles = roles.map(role => role.id);
+      }
+    }
+    if (options.accessToken) {
+      options.access_token = options.accessToken;
+      delete options.accessToken;
+    }
+    if (!options.access_token) return Promise.reject(new Error('User oauth2 access_token was not specified.'));
```

Comment:
change to `OAuth2 access token was not specified` and put the check on `options.accessToken`